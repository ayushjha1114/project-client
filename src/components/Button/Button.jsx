import React from 'react';
import PropTypes from 'prop-types';
import style from './style';

const propTypes = {
  color: PropTypes.string,
  disabled: PropTypes.bool,
  // onClick: PropTypes.func.isRequired,
  style: PropTypes.objectOf(PropTypes.string),
  value: PropTypes.string,
};

// default values for props:
const defaultProps = {
  value: '',
  color: '',
  disabled: false,
  style: {},
};

class Button extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    const {
      disabled,
      color,
      ...rest
    } = this.props;
    return (
      <>
        <input type="submit" disabled={disabled} {...rest} style={{ ...style.base, ...color }} />
      </>
    );
  }
}

Button.propTypes = propTypes;
Button.defaultProps = defaultProps;

export default Button;
