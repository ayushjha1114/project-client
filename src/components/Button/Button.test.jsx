import React from 'react';
import { shallow, mount } from 'enzyme';
import Button from './Button';

describe('Button', () => {
  test('should render correctly', () => {
    const component = shallow(<Button />);
    expect(component).toMatchSnapshot();
  });
  it('check the disabled props', () => {
    const props = {
      disabled: true,
    };

    const component = mount(<Button {...props} />);
    expect(component.prop('disabled')).toEqual(true);
  });
});
