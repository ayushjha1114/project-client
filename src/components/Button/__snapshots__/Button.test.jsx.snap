// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    color=""
    disabled={false}
    style={Object {}}
    value=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <input
        disabled={false}
        style={
          Object {
            "margin": "5px",
            "padding": "10px",
          }
        }
        type="submit"
        value=""
      />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "disabled": false,
        "style": Object {
          "margin": "5px",
          "padding": "10px",
        },
        "type": "submit",
        "value": "",
      },
      "ref": null,
      "rendered": null,
      "type": "input",
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <input
          disabled={false}
          style={
            Object {
              "margin": "5px",
              "padding": "10px",
            }
          }
          type="submit"
          value=""
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "disabled": false,
          "style": Object {
            "margin": "5px",
            "padding": "10px",
          },
          "type": "submit",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
